Automattic/jetpack
# My Jetpack: Add red bubble & notice when Protect threats detected

This PR adds the redbubble and notice/banner when Protect detects any vulnerability threats.





Fixes https://github.com/Automattic/jetpack-roadmap/issues/1431

### Other information:


## Jetpack product discussion



Project Thread: pbNhbs-bJN-p2
Design post: p1HpG7-umC-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:






- Install/activate Jetpack Protect with a paid Scan upgrade/plan
- Using the Scan Helper in the [Jetpack Debug Helper/Tools plugin](https://github.com/Automattic/jetpack-debug-helper/tree/trunk), select the first EIACAR Threats selection to Add five EICAR threats to the site (one for every severity).
- Run a scan to detect them - ensure threats are detected.
- Go to My Jetpack page.
- Verify you see a red bubble on the top level "Jetpack" menu item, and you see the notice/banner at the top of the page (See screenshot):

![Screen Shot 2024-12-21 at 10 17 55](https://github.com/user-attachments/assets/ddac15bf-5b5d-4918-9335-89c28361b5a7)

- Verify the notice/banner is an error notice (has a red border and a a red error icon).
- Now, using the Scan Helper in the Jetpack Debug Helper/Tools plugin again, remove the first EIACAR Threats selection  and add a different threat (like maybe a vulnerable plugin threat) - This one will be 'non-critical'.
- Run a new scan to detect the new threat changes.
- Now verify the notice/banner is an warning notice (has a yellow border and a yellow warning icon). See screenshot:

![Screen Shot on 2024-12-21 at 11-20-32](https://github.com/user-attachments/assets/b596c46c-3b39-4011-bece-c645469ad44c)

- Check the CTA's. Verify the Primary CTA leads to the Protect plugin dashboard, and the secondary CTA leads to the Jetpack.com `/support/scan/#how-do-i-fix-threats` page.
- Check the notice/banner in all viewport sizes.
- Disable the Protect standalone plugin and activate only the Main Jetpack plugin.
- Verify everything seems the same except the references to "Protect" now say "Scan", and that the primary CTA leads to the Scan dashboard in Jetpack cloud, instead of the Protect plugin. See screenshot:

![Screen Shot 2024-12-21 at 11 50 42](https://github.com/user-attachments/assets/c8b885fa-71fc-4c79-a76b-d0ca5d52427b)
